airflow/executors/base_executor.py

```diff
@@ -302,15 +319,40 @@ def trigger_tasks(self, open_slots: int) -> None:
                 if key in self.attempts:
                     del self.attempts[key]
                 task_tuples.append((key, command, queue, ti.executor_config))
+                if span.is_recording():
+                    span.add_event(
+                        name="task to trigger",
+                        attributes={"command": str(command), "conf": str(ti.executor_config)},
+                    )
 
         if task_tuples:
             self._process_tasks(task_tuples)
 
+    @span
     def _process_tasks(self, task_tuples: list[TaskTuple]) -> None:
         for key, command, queue, executor_config in task_tuples:
-            del self.queued_tasks[key]
-            self.execute_async(key=key, command=command, queue=queue, executor_config=executor_config)
-            self.running.add(key)
+            task_instance = self.queued_tasks[key][3]  # TaskInstance in fourth element
+            trace_id = int(gen_trace_id(task_instance.dag_run, as_int=True))
+            span_id = int(gen_span_id_from_ti_key(key, as_int=True))
+            links = [{"trace_id": trace_id, "span_id": span_id}]
+
+            # assuming that the span_id will very likely be unique inside the trace
+            with Trace.start_span(
+                span_name=f"{key.dag_id}.{key.task_id}",
+                component="BaseExecutor",
+                span_id=span_id,
+                links=links,
+            ) as span:
+                span.set_attribute("dag_id", key.dag_id)
+                span.set_attribute("run_id", key.run_id)
+                span.set_attribute("task_id", key.task_id)
+                span.set_attribute("try_number", key.try_number)
+                span.set_attribute("command", str(command))
```

Comment:
What is the value of putting the command in the span? It simply duplicates (in a format that is of no use as it's a single string) the dag_id, run_id, task_id etc.

Reply:
> What is the value of putting the command in the span? It simply duplicates (in a format that is of no use as it's a single string) the dag_id, run_id, task_id etc.

I did put the command as attribute, assuming that there may be something additional other than dag_id, run_id, task_id, etc. Due to the fact that I did not have too much of deep understanding of what the command can be, felt it was worth recording it as part of the span. If there's no real value of instrumenting the details of 'command' whatsoever, I'd say we can remove that instrumentation out of it for the next release.